pkg/pgsql/server/fmessages/bind.go

```diff
@@ -0,0 +1,154 @@
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fmessages
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+)
+
+// BindMsg Once a prepared statement exists, it can be readied for execution using a Bind message. The Bind message gives the
+// name of the source prepared statement (empty string denotes the unnamed prepared statement), the name of the destination
+// portal (empty string denotes the unnamed portal), and the values to use for any parameter placeholders present in the
+// prepared statement. The supplied parameter set must match those needed by the prepared statement. (If you declared
+// any void parameters in the Parse message, pass NULL values for them in the Bind message.) Bind also specifies
+// the format to use for any data returned by the query; the format can be specified overall, or per-column.
+// The response is either BindComplete or ErrorResponse.
+//
+// Note
+// The choice between text and binary output is determined by the format codes given in Bind, regardless of the SQL
+// command involved. The BINARY attribute in cursor declarations is irrelevant when using extended query protocol.
+type BindMsg struct {
+	// The name of the destination portal (an empty string selects the unnamed portal).
+	DestPortalName string
+	// The name of the source prepared statement (an empty string selects the unnamed prepared statement).
+	PreparedStatementName string
+	// The parameter format codes. Each must presently be zero (text) or one (binary).
+	ParameterFormatCodes []int16
+	// Array of the values of the parameters, in the format indicated by the associated format code. n is the above length.
+	ParamVals []interface{}
+	// The result-column format codes. Each must presently be zero (text) or one (binary).
+	ResultColumnFormatCodes []int16
+}
+
+func ParseBindMsg(payload []byte) (BindMsg, error) {
+	b := bytes.NewBuffer(payload)
+	r := bufio.NewReader(b)
+	destPortalName, err := getNextString(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+	preparedStatement, err := getNextString(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+	// The number of parameter format codes that follow (denoted C below).
+	// This can be zero to indicate that there are no parameters or that the parameters all use the default format (text);
+	// or one, in which case the specified format code is applied to all parameters; or it can equal the actual number
+	// of parameters.
+	parameterFormatCodeNumber, err := getNextInt16(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+	parameterFormatCodes := make([]int16, parameterFormatCodeNumber)
```

Comment:
`parameterFormatCodeNumber` can be negative here causing a panic. Please sanitize it.